gitcoin submission

vyomshm opened the pull request:
https://etherscan.io/tx/0xe9a8a0881abbd95ec078fe6bcbaa4662845fc954755ea52de431194f2cad43ba

RobKelley commented on contracts/Deposit.sol:
```diff
@@ -0,0 +1,84 @@
+pragma solidity ^0.4.19;
+
+contract AssetManager{
```
This should be "DepositManager"

RobKelley commented on contracts/Deposit.sol:
```diff
@@ -0,0 +1,84 @@
+        creator = msg.sender;
+    }
+    
+    function createDeposit(address assetOwner, uint fee, uint daysUntilExpiry) payable public{
```
in the scaled back implementation, **fee** should a transaction fee paid to cinch upon initialization of the deposit.  It's not passed by the depositmaker.  It can be hardcoded within the smart contract. 
the asset rental fee is beyond the scope of the first version.

RobKelley commented on contracts/Deposit.sol:
```diff
@@ -0,0 +1,84 @@
+        depositor = _depositor;
+        assetOwner = _assetOwner;
+        fee = _fee;
+        timeOfCreation = now; 
```
I got a warning on compile:
"browser/SimpleDeposit.sol:55:26:use of "now": "now" does not mean current time. Now is an alias for block.timestamp. Block.timestamp can be influenced by miners to a certain degree, be careful."

Just FYI.

RobKelley commented on contracts/Deposit.sol:
```diff
@@ -0,0 +1,84 @@
+    }
+    
+    // depositor calls this function after assetInfo has been shared and expiry time is past
+    function retrieveDeposit() onlyDepositor public payable{
```
I got a compile error as well as ongoing showstopper problems setting a deposit, with warnings about Gas:

"Gas requirement of function SimpleDeposit.retrieveDeposit() high: infinite. If the gas requirement of a function is higher than the block gas limit, it cannot be executed. Please avoid loops in your functions or actions that modify large areas of storage (this includes clearing or copying arrays in storage)"

vyomshm commented:
Hey Rob, 
Thanks for your input, I've fixed some bugs and made some updates.

1) updated value inputs to be set in Wei. This was the reason behind failing deposits.
2) changed `AssetManager` to `DepositManager`
3) I'm aware of the warnings generated when compiling the contract - Although using an Oracle for date-time purposes is also an option, using `now` should be perfectly safe and sufficient for the purposes of this contract. Also the warnings about gas are purely perfunctory, Remix shows them for all functions modifying storage.
4) About the fee value, `fee` / `feeInWei` var represents the fee to be paid to the assetOwner after he/she has shared the required `assetInfo`.

If you'd like to have a third value (`cinchFee`) to be paid to a different `CinchAddress`, that can also be achieved by adding few lines of code to the `SimpleDeposit` contract - 

`address cinch = 0x72ba7d8e73fe8eb666ea66babc8116a41bfb10e2;
uint cinchFee = 0.005 ether;`

and updating `retrieveDeposit()` function to - 

`
function retrieveDeposit() onlyDepositor public payable{
        require(now > timeOfCreation + validFor * 1 hours);
        require(assetInfo != bytes32(0));
        assetOwner.transfer(fee);
        cinch.transfer(cinchFee);
        selfdestruct(depositor);
    }
`

RobKelley commented on contracts/README.md:
```diff
@@ -0,0 +1,51 @@
+
+Note - all fee values should be in Wei( 1 ether = 1,000,000,000,000,000,000 Wei) (convertor- https://etherconverter.online/ or web3.utils.toWei())
+
+* call the createDeposit function - `createDeposit(0xca35b7d915458ef540ade6068dfe2f44e8fa733c, 200000000000000000, 10)`   with the transaction value set to greator than or equal to the fee value ( in this example - value >= 0.4 ether)
```
OK, still getting errors calling createDeposit following the method in the video, but with the values "200000000000000000, 10" after the AssetOwner's address.   Any idea why:
https://rinkeby.etherscan.io/tx/0xcc2d2bc6a2b7c2b6cf80d13d7c61797420491c4804b9be181224139592fd9b6e

If I could reliably create deposits in this contract, I'd be satisfied.

vyomshm replied:
I noticed that your tx value https://rinkeby.etherscan.io/tx/0xcc2d2bc6a2b7c2b6cf80d13d7c61797420491c4804b9be181224139592fd9b6e is set to `0`. 
To create a deposit, you need to send the deposit amount along with the transaction. This is a cost effective solution and saves us gas by reducing the total number of transactions. To call the `createDeposit()` function, you need to send along the ether deposit.
For example - if I set the `feeInWei = 200000000000000000`(i.e. 0.2 ether), then in this case, set the tx "value" to greater than or equal to 0.4 ether(i.e. twice the `feeInWei`). 

A successful tx will look like https://rinkeby.etherscan.io/tx/0x315cd03c2b40fc359cca315f2d6971facfc6ae001b664f16ac3dc072c173f4c6 , notice the difference in the value field.

RobKelley commented:
I don't think I'm sending zero.   It's failing to zero.   I'm sending
createDeposit with the [deposit-taker address], 200000000000000000, 10

Is there a specific place in the Remix IDE I set tx value?


On Mon, Apr 23, 2018 at 1:16 PM, Vyom Sharma <notifications@github.com>
wrote:

> *@vyomshm* commented on this pull request.
> ------------------------------
>
> In contracts/README.md
> <https://github.com/RobKelley/its-a-cinch/pull/2#discussion_r183472318>:
>
> > @@ -0,0 +1,51 @@
> +
> +## Workflow - how to create a SimpledEposit using a running DepositManager Contract
> +
> +Note - all fee values should be in Wei( 1 ether = 1,000,000,000,000,000,000 Wei) (convertor- https://etherconverter.online/ or web3.utils.toWei())
> +
> +* call the createDeposit function - `createDeposit(0xca35b7d915458ef540ade6068dfe2f44e8fa733c, 200000000000000000, 10)`   with the transaction value set to greator than or equal to the fee value ( in this example - value >= 0.4 ether)
>
> I noticed that your tx value https://rinkeby.etherscan.io/t
> x/0xcc2d2bc6a2b7c2b6cf80d13d7c61797420491c4804b9be181224139592fd9b6e is
> set to 0.
> To create a deposit, you need to send the deposit amount along with the
> transaction. This is a cost effective solution and saves us gas by reducing
> the total number of transactions. To call the createDeposit() function,
> you need to send along the ether deposit.
> For example - if I set the feeInWei = 200000000000000000(i.e. 0.2 ether),
> then in this case, set the tx "value" to greater than or equal to 0.4
> ether(i.e. twice the feeInWei).
>
> A successful tx will look like https://rinkeby.etherscan.io/t
> x/0x315cd03c2b40fc359cca315f2d6971facfc6ae001b664f16ac3dc072c173f4c6 ,
> notice the difference in the value field.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/RobKelley/its-a-cinch/pull/2#discussion_r183472318>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAI4NtsCycJRoHrlgr8EjSh1vBzzoKczks5trgxcgaJpZM4TVP41>
> .
>

vyomshm commented:
Yes, there's a specific place to set the tx value. Notice the field marked
"value" at the top right section, below the "Account" and "Gas Limit". 
![screen shot 2018-04-23 at 10 42 35 pm](https://user-images.githubusercontent.com/20184946/39143018-1d113684-474a-11e8-9714-a48331c5a504.png)


On Mon, Apr 23, 2018 at 10:55 PM, Rob <notifications@github.com> wrote:

> I don't think I'm sending zero. It's failing to zero. I'm sending
> createDeposit with the [deposit-taker address], 200000000000000000, 10
>
> Is there a specific place in the Remix IDE I set tx value?
>
>
> On Mon, Apr 23, 2018 at 1:16 PM, Vyom Sharma <notifications@github.com>
> wrote:
>
> > *@vyomshm* commented on this pull request.
> > ------------------------------
> >
> > In contracts/README.md
> > <https://github.com/RobKelley/its-a-cinch/pull/2#discussion_r183472318>:
> >
> > > @@ -0,0 +1,51 @@
> > +
> > +## Workflow - how to create a SimpledEposit using a running
> DepositManager Contract
> > +
> > +Note - all fee values should be in Wei( 1 ether =
> 1,000,000,000,000,000,000 Wei) (convertor- https://etherconverter.online/
> or web3.utils.toWei())
> > +
> > +* call the createDeposit function - `createDeposit(
> 0xca35b7d915458ef540ade6068dfe2f44e8fa733c, 200000000000000000, 10)` with
> the transaction value set to greator than or equal to the fee value ( in
> this example - value >= 0.4 ether)
> >
> > I noticed that your tx value https://rinkeby.etherscan.io/t
> > x/0xcc2d2bc6a2b7c2b6cf80d13d7c61797420491c4804b9be181224139592fd9b6e is
> > set to 0.
> > To create a deposit, you need to send the deposit amount along with the
> > transaction. This is a cost effective solution and saves us gas by
> reducing
> > the total number of transactions. To call the createDeposit() function,
> > you need to send along the ether deposit.
> > For example - if I set the feeInWei = 200000000000000000(i.e. 0.2 ether),
> > then in this case, set the tx "value" to greater than or equal to 0.4
> > ether(i.e. twice the feeInWei).
> >
> > A successful tx will look like https://rinkeby.etherscan.io/t
> > x/0x315cd03c2b40fc359cca315f2d6971facfc6ae001b664f16ac3dc072c173f4c6 ,
> > notice the difference in the value field.
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/RobKelley/its-a-cinch/pull/2#discussion_r183472318>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/
> AAI4NtsCycJRoHrlgr8EjSh1vBzzoKczks5trgxcgaJpZM4TVP41>
>
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/RobKelley/its-a-cinch/pull/2#issuecomment-383655565>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATP_crIEVIsbNgDpNbs2VXQIHFoOwl1Xks5trg6NgaJpZM4TVP41>
> .
>

vyomshm commented:
To demonstrate, I just created a deposit of `0.7 ether` to your account(
`0x9195eD493E9c7359DD963a83a68C1645001bd4E3` ) with fee value set = `0.3 ether`
, using your deployed `DepositManager` Contract located at
`0x4786d8ddb7c6b665be56673250192a544900d1b7`
https://rinkeby.etherscan.io/address/0x4786d8ddb7c6b665be56673250192a544900d1b7



Tx -
https://rinkeby.etherscan.io/tx/0x1cb91b31472669cd7065503b136ccd0617bec79723330165eb9fd37a21b88531

You can see find the resulting SimpleDeposit Contract at
`0x0f4b067315343b521f0c20513d5854c1fb0a95a8`



On Mon, Apr 23, 2018 at 10:59 PM, Vyom Sharma <vyomshm@gmail.com> wrote:

> Yes, there's a specific place to set the tx value. Notice the field marked
> "value" at the top right section, below the "Account" and "Gas Limit"
>
> On Mon, Apr 23, 2018 at 10:55 PM, Rob <notifications@github.com> wrote:
>
>> I don't think I'm sending zero. It's failing to zero. I'm sending
>> createDeposit with the [deposit-taker address], 200000000000000000, 10
>>
>> Is there a specific place in the Remix IDE I set tx value?
>>
>>
>> On Mon, Apr 23, 2018 at 1:16 PM, Vyom Sharma <notifications@github.com>
>> wrote:
>>
>> > *@vyomshm* commented on this pull request.
>> > ------------------------------
>> >
>> > In contracts/README.md
>> > <https://github.com/RobKelley/its-a-cinch/pull/2#discussion_r183472318
>> >:
>> >
>> > > @@ -0,0 +1,51 @@
>> > +
>> > +## Workflow - how to create a SimpledEposit using a running
>> DepositManager Contract
>> > +
>> > +Note - all fee values should be in Wei( 1 ether =
>> 1,000,000,000,000,000,000 Wei) (convertor- https://etherconverter.online/
>> or web3.utils.toWei())
>> > +
>> > +* call the createDeposit function - `createDeposit(0xca35b7d915458ef540ade6068dfe2f44e8fa733c,
>> 200000000000000000, 10)` with the transaction value set to greator than or
>> equal to the fee value ( in this example - value >= 0.4 ether)
>> >
>> > I noticed that your tx value https://rinkeby.etherscan.io/t
>> > x/0xcc2d2bc6a2b7c2b6cf80d13d7c61797420491c4804b9be181224139592fd9b6e is
>> > set to 0.
>> > To create a deposit, you need to send the deposit amount along with the
>> > transaction. This is a cost effective solution and saves us gas by
>> reducing
>> > the total number of transactions. To call the createDeposit() function,
>> > you need to send along the ether deposit.
>> > For example - if I set the feeInWei = 200000000000000000(i.e. 0.2
>> ether),
>> > then in this case, set the tx "value" to greater than or equal to 0.4
>> > ether(i.e. twice the feeInWei).
>> >
>> > A successful tx will look like https://rinkeby.etherscan.io/t
>> > x/0x315cd03c2b40fc359cca315f2d6971facfc6ae001b664f16ac3dc072c173f4c6 ,
>> > notice the difference in the value field.
>> >
>> > —
>> > You are receiving this because you commented.
>> > Reply to this email directly, view it on GitHub
>> > <https://github.com/RobKelley/its-a-cinch/pull/2#discussion_r183472318
>> >,
>> > or mute the thread
>> > <https://github.com/notifications/unsubscribe-auth/AAI4NtsCy
>> cJRoHrlgr8EjSh1vBzzoKczks5trgxcgaJpZM4TVP41>
>>
>> > .
>> >
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/RobKelley/its-a-cinch/pull/2#issuecomment-383655565>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ATP_crIEVIsbNgDpNbs2VXQIHFoOwl1Xks5trg6NgaJpZM4TVP41>
>> .
>>
>
>

RobKelley commented:
OK, Ill try this tonight

On Mon, Apr 23, 2018 at 1:29 PM, Vyom Sharma <notifications@github.com>
wrote:

> Yes, there's a specific place to set the tx value. Notice the field marked
> "value" at the top right section, below the "Account" and "Gas Limit"
>
> On Mon, Apr 23, 2018 at 10:55 PM, Rob <notifications@github.com> wrote:
>
> > I don't think I'm sending zero. It's failing to zero. I'm sending
> > createDeposit with the [deposit-taker address], 200000000000000000, 10
> >
> > Is there a specific place in the Remix IDE I set tx value?
> >
> >
> > On Mon, Apr 23, 2018 at 1:16 PM, Vyom Sharma <notifications@github.com>
> > wrote:
> >
> > > *@vyomshm* commented on this pull request.
> > > ------------------------------
> > >
> > > In contracts/README.md
> > > <https://github.com/RobKelley/its-a-cinch/pull/2#discussion_r183472318
> >:
> > >
> > > > @@ -0,0 +1,51 @@
> > > +
> > > +## Workflow - how to create a SimpledEposit using a running
> > DepositManager Contract
> > > +
> > > +Note - all fee values should be in Wei( 1 ether =
> > 1,000,000,000,000,000,000 Wei) (convertor-
> https://etherconverter.online/
> > or web3.utils.toWei())
> > > +
> > > +* call the createDeposit function - `createDeposit(
> > 0xca35b7d915458ef540ade6068dfe2f44e8fa733c, 200000000000000000, 10)`
> with
> > the transaction value set to greator than or equal to the fee value ( in
> > this example - value >= 0.4 ether)
> > >
> > > I noticed that your tx value https://rinkeby.etherscan.io/t
> > > x/0xcc2d2bc6a2b7c2b6cf80d13d7c61797420491c4804b9be181224139592fd9b6e
> is
> > > set to 0.
> > > To create a deposit, you need to send the deposit amount along with the
> > > transaction. This is a cost effective solution and saves us gas by
> > reducing
> > > the total number of transactions. To call the createDeposit() function,
> > > you need to send along the ether deposit.
> > > For example - if I set the feeInWei = 200000000000000000(i.e. 0.2
> ether),
> > > then in this case, set the tx "value" to greater than or equal to 0.4
> > > ether(i.e. twice the feeInWei).
> > >
> > > A successful tx will look like https://rinkeby.etherscan.io/t
> > > x/0x315cd03c2b40fc359cca315f2d6971facfc6ae001b664f16ac3dc072c173f4c6 ,
> > > notice the difference in the value field.
> > >
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/RobKelley/its-a-cinch/pull/2#discussion_r183472318
> >,
> > > or mute the thread
> > > <https://github.com/notifications/unsubscribe-auth/
> > AAI4NtsCycJRoHrlgr8EjSh1vBzzoKczks5trgxcgaJpZM4TVP41>
> >
> > > .
> > >
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/RobKelley/its-a-cinch/pull/2#issuecomment-383655565
> >,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/ATP_
> crIEVIsbNgDpNbs2VXQIHFoOwl1Xks5trg6NgaJpZM4TVP41>
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/RobKelley/its-a-cinch/pull/2#issuecomment-383656848>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAI4NvGOhjML14WDd8fW83e6A0laRMMnks5trg98gaJpZM4TVP41>
> .
>

RobKelley commented:
Success!

https://rinkeby.etherscan.io/address/0x9195ed493e9c7359dd963a83a68c1645001bd4e3

On Mon, Apr 23, 2018 at 4:27 PM, Rob Kelley <rob@robkelley.net> wrote:

> OK, Ill try this tonight
>
> On Mon, Apr 23, 2018 at 1:29 PM, Vyom Sharma <notifications@github.com>
> wrote:
>
>> Yes, there's a specific place to set the tx value. Notice the field marked
>> "value" at the top right section, below the "Account" and "Gas Limit"
>>
>> On Mon, Apr 23, 2018 at 10:55 PM, Rob <notifications@github.com> wrote:
>>
>> > I don't think I'm sending zero. It's failing to zero. I'm sending
>> > createDeposit with the [deposit-taker address], 200000000000000000, 10
>> >
>> > Is there a specific place in the Remix IDE I set tx value?
>> >
>> >
>> > On Mon, Apr 23, 2018 at 1:16 PM, Vyom Sharma <notifications@github.com>
>> > wrote:
>> >
>> > > *@vyomshm* commented on this pull request.
>> > > ------------------------------
>> > >
>> > > In contracts/README.md
>> > > <https://github.com/RobKelley/its-a-cinch/pull/2#discussion_
>> r183472318>:
>> > >
>> > > > @@ -0,0 +1,51 @@
>> > > +
>> > > +## Workflow - how to create a SimpledEposit using a running
>> > DepositManager Contract
>> > > +
>> > > +Note - all fee values should be in Wei( 1 ether =
>> > 1,000,000,000,000,000,000 Wei) (convertor-
>> https://etherconverter.online/
>> > or web3.utils.toWei())
>> > > +
>> > > +* call the createDeposit function - `createDeposit(
>> > 0xca35b7d915458ef540ade6068dfe2f44e8fa733c, 200000000000000000, 10)`
>> with
>> > the transaction value set to greator than or equal to the fee value ( in
>> > this example - value >= 0.4 ether)
>> > >
>> > > I noticed that your tx value https://rinkeby.etherscan.io/t
>> > > x/0xcc2d2bc6a2b7c2b6cf80d13d7c61797420491c4804b9be181224139592fd9b6e
>> is
>> > > set to 0.
>> > > To create a deposit, you need to send the deposit amount along with
>> the
>> > > transaction. This is a cost effective solution and saves us gas by
>> > reducing
>> > > the total number of transactions. To call the createDeposit()
>> function,
>> > > you need to send along the ether deposit.
>> > > For example - if I set the feeInWei = 200000000000000000(i.e. 0.2
>> ether),
>> > > then in this case, set the tx "value" to greater than or equal to 0.4
>> > > ether(i.e. twice the feeInWei).
>> > >
>> > > A successful tx will look like https://rinkeby.etherscan.io/t
>> > > x/0x315cd03c2b40fc359cca315f2d6971facfc6ae001b664f16ac3dc072c173f4c6
>> ,
>> > > notice the difference in the value field.
>> > >
>> > > —
>> > > You are receiving this because you commented.
>> > > Reply to this email directly, view it on GitHub
>> > > <https://github.com/RobKelley/its-a-cinch/pull/2#discussion_
>> r183472318>,
>> > > or mute the thread
>> > > <https://github.com/notifications/unsubscribe-auth/
>> > AAI4NtsCycJRoHrlgr8EjSh1vBzzoKczks5trgxcgaJpZM4TVP41>
>> >
>> > > .
>> > >
>> >
>> > —
>> > You are receiving this because you authored the thread.
>> > Reply to this email directly, view it on GitHub
>> > <https://github.com/RobKelley/its-a-cinch/pull/2#issuecomment-383655565
>> >,
>> > or mute the thread
>> > <https://github.com/notifications/unsubscribe-auth/ATP_crIEV
>> IsbNgDpNbs2VXQIHFoOwl1Xks5trg6NgaJpZM4TVP41>
>> > .
>> >
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/RobKelley/its-a-cinch/pull/2#issuecomment-383656848>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAI4NvGOhjML14WDd8fW83e6A0laRMMnks5trg98gaJpZM4TVP41>
>> .
>>
>
>
